src/tools/illink/src/linker/Linker/TypeMapInfo.cs

```diff
@@ -257,17 +261,20 @@ void MapOverrides (MethodDefinition method)
 				if (baseMethod == null)
 					continue;
 				if (baseMethod.DeclaringType.IsInterface) {
-					AnnotateMethods (baseMethod, method, InterfaceImplementor.Create (method.DeclaringType, baseMethod.DeclaringType, context));
+					var found = _runtimeInterfaceImpls.TryGetValue (method.DeclaringType, out var runtimeInterfaces);
+					Debug.Assert (found);
+					var runtimeInterface = runtimeInterfaces.Single (i => TypeReferenceEqualityComparer.AreEqual (i.InflatedInterfaceType, baseMethodRef.DeclaringType, context));
```

Comment:
I don't think a methodImpl will always be for an explicit interface - this will check all the runtime interfaces. Would you rather have a Debug.Assert that there's only one runtime interface that matches, and choose the first instead?